backend/api/views/quiz.py

```diff
@@ -271,23 +272,51 @@ def create_quiz_result():
         )
 
 
+def delete_quiz_by_id(user_data):
+    if invalid_model_helper(user_data, ["quiz_id"]):
+        return False
+
+    quiz_to_delete = Quiz.query.get(user_data["quiz_id"])
+    if quiz_to_delete is None:
+        return False
+
+    db.session.delete(quiz_to_delete)
+    db.session.commit()
+    return True
+
+
 @quiz.route("/delete_quiz", methods=["POST"])
 @admin_route
 def delete_quiz(user_id):
     user_data = request.get_json()
-    if invalid_model_helper(user_data, ["quiz_id"]):
+    did_delete = delete_quiz_by_id(user_data)
+    if not did_delete:
         return create_response(
-            message="Missing quiz id", status=422, data={"status": "fail"}
+            message="Quiz id missing or quiz not found",
+            status=422,
+            data={"status": "fail"},
         )
 
-    quiz_to_delete = Quiz.query.get(user_data["quiz_id"])
-    if quiz_to_delete is None:
+    return create_response(
+        message="Successfully created quiz result; new badges earned",
+        status=200,
+        data={"status": "success"},
+    )
+
+
+@quiz.route("/edit_quiz", methods=["POST"])
+def edit_quiz():
+    pdb.set_trace()
+    user_data = request.get_json()
+    did_delete = delete_quiz_by_id(user_data)
+    if not did_delete:
         return create_response(
-            message="Quiz not found", status=422, data={"status": "fail"}
+            message="Quiz id missing or quiz not found",
+            status=422,
+            data={"status": "fail"},
         )
 
-    db.session.delete(quiz_to_delete)
-    db.session.commit()
+    res = create_quiz_helper(request.get_json())
     return create_response(
-        message="Successfully deleted quiz", status=200, data={"status": "success"}
+        message=res["message"], status=res["status"], data=res["data"]
```

Comment:
Might want to switch message from successfully created quiz to successfully edited quiz here